common/src/test/java/org/opensearch/ml/common/transport/model/MLUpdateModelInputTest.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.common.transport.model;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.function.Consumer;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.opensearch.common.io.stream.BytesStreamOutput;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.xcontent.LoggingDeprecationHandler;
+import org.opensearch.common.xcontent.XContentFactory;
+import org.opensearch.common.xcontent.XContentType;
+import org.opensearch.core.xcontent.NamedXContentRegistry;
+import org.opensearch.core.xcontent.ToXContent;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.search.SearchModule;
+import org.opensearch.ml.common.model.MLModelConfig;
+import org.opensearch.ml.common.model.TextEmbeddingModelConfig;
+
+public class MLUpdateModelInputTest {
+
+    private MLUpdateModelInput updateModelInput;
+    private final String expectedInputStr = "{\"model_id\":\"test-model_id\",\"description\":\"description\",\"name\":\"name\",\"model_group_id\":\"modelGroupId\",\"model_config\":" +
```

Comment:
No we can't. The model ID is not allowed to update, if you sent an update request like this it will directly ignore the model_id in your request body and will only use the model_id you provide in your API endpoint.

Reply:
Seems not necessary to put model id in request body, similar to update detectyor API, https://opensearch.org/docs/latest/observing-your-data/ad/api/#update-detector `PUT _plugins/_anomaly_detection/detectors/<detectorId>` , we can use `PUT _plugins/_ml/models/<model_Id>`

Reply:
Besides, we used POST method here due to implementing OS built-in update method.  https://opensearch.org/docs/latest/api-reference/document-apis/update-document/